spec/integration/startup-spec.coffee

```diff
@@ -241,6 +241,42 @@ describe "Starting Atom", ->
                 [otherTempDirPath]
               ].sort()
 
+    it "doesn't reopen any previously opened windows if restorePreviousWindowsOnStart is disabled", ->
+      configPath = path.join(atomHome, 'config.cson')
+      config = CSON.readFileSync(configPath)
+      config['*'].core = {restorePreviousWindowsOnStart: false}
+      CSON.writeFileSync(configPath, config)
+
+      runAtom [tempDirPath], {ATOM_HOME: atomHome}, (client) ->
+        client
+          .waitForExist("atom-workspace")
+          .waitForNewWindow(->
+            @startAnotherAtom([otherTempDirPath], ATOM_HOME: atomHome)
+          , 5000)
+          .waitForExist("atom-workspace")
+
+      runAtom [], {ATOM_HOME: atomHome}, (client) ->
+        windowProjectPaths = []
+
+        client
+          .waitForWindowCount(2, 10000)
+          .then ({value: windowHandles}) ->
+            @window(windowHandles[0])
+            .waitForExist("atom-workspace")
+            .treeViewRootDirectories()
+            .then ({value: directories}) -> windowProjectPaths.push(directories)
+
+            .window(windowHandles[1])
+            .waitForExist("atom-workspace")
+            .treeViewRootDirectories()
+            .then ({value: directories}) -> windowProjectPaths.push(directories)
+
+            .call ->
+              expect(windowProjectPaths).toEqual [
+                []
+                []
+              ]
```

Comment:
Nice work adding this integration test. The assertion doesn't seem right to me though; this is saying that there should be *two* Atom windows, each with *no* project paths. If we've chosen to *not* restore the session though, shouldn't there just be *one* Atom window?